torch/csrc/jit/ir.cpp

```diff
@@ -1652,10 +1669,24 @@ at::ArrayRef<Value*> createTupleUnpack(Value* v) {
 
 std::vector<Value*> inlineCallTo(Node* to_replace, Function* callee) {
   WithInsertPoint guard(to_replace);
+  std::unordered_map<Value*, Value*> value_map;
   auto new_outputs = insertGraph(
       *to_replace->owningGraph(),
       *(callee->optimized_graph()),
-      to_replace->inputs());
+      to_replace->inputs(),
+      value_map);
+
+  // TODO: We might need to use nodes_map instead of value_map. Otherwise, we
+  // are missing nodes without outputs (e.g. prim::Print).
+  std::unordered_set<Node*> updated_nodes;
+  for (const auto& kv : value_map) {
+    Node* orig_node = kv.first->node();
+    Node* new_node = kv.second->node();
+    if (updated_nodes.insert(new_node).second) {
+      new_node->insertCallStackEntry(callee, to_replace->sourceRange());
```

Comment:
I believe insertGraph preserves source ranges. But it looks like it does not preserve callstacks as well? That seems inconsistent. My expectation would be that it would not drop part of the information about a node.  It might make more sense for insertGraph to take an optional `callee_function` and have it do the bookkeeping to update the graph, otherwise it would copy the callstack information as is. The current algorithm has the Print issue as documented above, and is nondeterministic in the order it processes nodes. This isn't wrong, but will make debugging errors around this code harder.

Reply:
I changed how we update the callstacks in insertGraph : now when we copy a node we copy its callstack as well and then here we just need to add one more entry to it (that also allowed me to delete "appendCallstack.." function).